dffml/service/dev.py

```diff
@@ -1056,6 +1060,96 @@ class Bump(CMD):
     packages = BumpPackages
 
 
+class SphinxBuildError(Exception):
+    """
+    Raised when sphinx-build command exits with a non-zero error code.
+    """
+
+    def __str__(self):
+        return "[ERROR] Failed run sphinx, is it installed (pip install -U .[dev])?"
+
+
+class MakeDocs(CMD):
+    """
+    Generate docs of the complete package 
+    """
+
+    async def run(self):
+
+        root = Path(__file__).parents[2]
+
+        pages_path = root / "pages"
+
+        shutil.rmtree(pages_path, ignore_errors=True)
+
+        docs_path = root / "docs"
+
+        files_to_check = [
+            ("changelog.md", "CHANGELOG.md"),
+            ("shouldi.md", "examples/shouldi/README.md"),
+            ("swportal.rst", "examples/swportal/README.rst"),
+            (
+                "contributing/consoletest.md",
+                "dffml/util/testing/consoletest/README.md",
+            ),
```

Comment:
Let's make `files_to_check` a list of tuples where each entry in each tuple is itself a tuple of the path components. This will allow us to use the  OS agnostic `pathlib` methods to reference for the correct files/links.

So the strings will become tuples split on `/`

For example:

```suggestion
            (("changelog.md",), ("CHANGELOG.md",)),
            (("shouldi.md",), ("examples", "shouldi", "README.md")),
            (("swportal.rst",), ("examples", "swportal", "README.rst")),
            (
                ("contributing", "consoletest.md"),
                ("dffml", "util", "testing", "consoletest", "README.md"),
            ),
```

Reply:
Yes , I din't have this in mind while writing the code that some `pathlib` methods are not OS agnostic , however I have changed this as per your comments